### PR TITLE
Fixing Guava 1635 issue with spark cassandra connector + other minor …

### PR DESCRIPTION
**Pull request by sreeraaman**
This PR fixes 4 issues as noted below:

1. Currently, the spark jobserver does not work nicely with the spark cassandra connector because of the Guava issue #736 because of the Cassandra driver leaking Guava 16+ whereas spark uses Guava 14+.

The resolution was to remove cassandra-drive-core and cassandra-driver-mapping and instead use spark-cassandra-connector as the dependency.

2. While building a docker image for spark 2.0+ from source tree, there seems to be an issue with the make_distribution.sh that got moved into a separate folder /dev under the source tree. However, for spark 1.6+, the make_distribution.sh seems to be at the root of the source tree. 

This has been fixed with a if else construct by checking the Versions.spark value.

3. Even after adding the spark-cassandra-connector as a dependency, it was not getting pulled because of the exclusion list in Assembly which had spark as a filter.

This was fixed by removing the spark filter from the filter list.

4. Finally while building a docker image, an error occurs after tagging where it detects the docker version and with my version of docker (Version 17.06.0-ce-mac19 (18663)), the sbt docker plugins fails with an error unable to parse version.

Looks like this was fixed in version 1.4.1.

5. Finally, a minor change was to bump the spark version from the default 2.1.0 to 2.1.1

---
This change is [Reviewable](https://reviewable.io/reviews/spark-jobserver/spark-jobserver/903)

### CODE REVIEW

File: job-server/config/docker.conf [hunk 1]
```diff
@@ -33,13 +33,14 @@ spark {
...
 
   # predefined Spark contexts
-  # contexts {
-  #   my-low-latency-context {
-  #     num-cpu-cores = 1           # Number of cores to allocate.  Required.
-  #     memory-per-node = 512m         # Executor memory per node, -Xmx style eg 512m, 1G, etc.
-  #   }
-  #   # define additional contexts here
-  # }
+   contexts {
```
Comment (noorul) [inline job-server/config/docker.conf]: Why are we setting the default to 4G?

File: job-server/config/local.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (noorul) [inline job-server/config/local.sh]: I think you should change the template file.

File: project/Assembly.scala [hunk 3]
```diff
@@ -5,12 +5,16 @@ import sbtassembly.AssemblyPlugin.autoImport._
 object Assembly {
   lazy val settings = Seq(
     assemblyJarName in assembly := "spark-job-server.jar",
-    // uncomment below to exclude tests
+/*
+    assemblyShadeRules in assembly := Seq(ShadeRule.rename("com.google.**" -> "sjs.com.google.@1").inAll),
```
Comment (noorul) [inline project/Assembly.scala]: Why are you adding commented code?

File: project/Assembly.scala [hunk 4]
```diff
@@ -5,12 +5,16 @@ import sbtassembly.AssemblyPlugin.autoImport._
...
+*/
+
+      // uncomment below to exclude tests
     // test in assembly := {},
     assemblyExcludedJars in assembly <<= (fullClasspath in assembly) map { _ filter { cp =>
       List("servlet-api", "guice-all", "junit", "uuid",
         "jetty", "jsp-api-2.0", "antlr", "avro", "slf4j-log4j", "log4j-1.2",
-        "scala-actors", "spark", "commons-cli", "stax-api", "mockito").exists(cp.data.getName.startsWith(_))
```
Comment (noorul) [inline project/Assembly.scala]: I am not sure this is required. Can we use master=local without this?

File: project/Dependencies.scala [hunk 5]
```diff
@@ -55,9 +55,12 @@ object Dependencies {
     "org.flywaydb" % "flyway-core" % flyway
   )
 
+
+
   lazy val cassandraDeps = Seq(
-    "com.datastax.cassandra" % "cassandra-driver-core" % cassandra,
-    "com.datastax.cassandra" % "cassandra-driver-mapping" % cassandra
+    /*"com.datastax.cassandra" % "cassandra-driver-core" % cassandra,
+    "com.datastax.cassandra" % "cassandra-driver-mapping" % cassandra*/
```
Comment (noorul) [inline project/Dependencies.scala]: Why do we need commented code?

**Comment by noorul** [comment]
Can you remove .DS_Store file?

File: project/Versions.scala [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import scala.util.Properties.isJavaAtLeast
 
 object Versions {
-  lazy val spark = sys.env.getOrElse("SPARK_VERSION", "2.1.0")
+  lazy val spark = sys.env.getOrElse("SPARK_VERSION", "2.1.1")
```
Comment (noorul) [inline project/Versions.scala]: Latest is 2.2.0, can we go to that?

File: build.sbt [hunk 7]
```diff
@@ -137,7 +137,13 @@ lazy val dockerSettings = Seq(
     val sparkBuild = s"spark-${Versions.spark}"
     val sparkBuildCmd = scalaBinaryVersion.value match {
       case "2.11" =>
-        "./make-distribution.sh -Dscala-2.11 -Phadoop-2.7 -Phive"
+        if (Versions.spark.startsWith("1")){
+          "./make-distribution.sh -Dscala-2.11 -Phadoop-2.7 -Phive"
+        }
+        else {
```
Comment (zeitos) [inline build.sbt]: we usually do
```
} else {
```

also: consider doing 
```
Versions.spark match {
   case s if s.startsWith("1") => {}
   case _ => {}
}
```

**Comment by sreeraaman** [comment]
@noorul. I have applied the changes as requested.

**Comment by noorul** [comment]
@sreeraaman You need to re-base as C* connector version that we use has changed in master.

**Comment by velvia** [comment]
This looks fine to me.   As long as @noorul is happy  :)

**Comment by velvia** [comment]
@sreeraaman are you still working on this PR?

File: .gitignore [hunk 8]
```diff
@@ -9,6 +9,7 @@ test-reports/
 .idea*
 .vagrant
 *.pyc
+.DS_Store
```
Comment (noorul) [inline .gitignore]: This should be in your global gitignore.

File: project/Assembly.scala [hunk 9]
```diff
@@ -5,12 +5,12 @@ import sbtassembly.AssemblyPlugin.autoImport._
 object Assembly {
   lazy val settings = Seq(
     assemblyJarName in assembly := "spark-job-server.jar",
-    // uncomment below to exclude tests
+      // uncomment below to exclude tests
```
Comment (noorul) [inline project/Assembly.scala]: This change is not required.

**Comment by zeitos** [comment]
LGTM with @noorul 's comments

**Pull request merged**